sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1037,6 +1057,10 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
             failOnInvalidOuterReference(j)
             failOnOuterReferenceInSubTree(left)
 
+          // Lateral join is not supported inside a correlated subquery.
+          case LateralJoin(_) =>
+            failOnOuterReferenceInSubTree(j)
```

Comment:
what's the error message? Do we clearly mention that lateral join inside correlated subquery is not allowed?

Reply:
The error message here is "Accessing outer query column is not allowed in LateralJoin...". I will make the error message more explicit.